doc/design/dist/README.md

```diff
@@ -0,0 +1,190 @@
+# Distributed Training Design Doc
+
+## Objective
+
+We want Paddle to support training on the general-purpose cluster. The cluster runs Paddle, the web server (e.g., Nginx), the log collector (e.g., fluentd), the distributed queue service (e.g., Kafka), the log joiner and other data processors written using Storm, Spark, and Hadoop MapReduce on the same cluster. As illustrated in the following graph:
+
+<img src="src/arch.png"/>
+
+This poses new challenges for Paddle,
+
+- Paddle need to be fault tolerant.
+- Input training data can be online data from real time logs or batch data from distributed file system.
+- User needs a simple way to train model on Paddle cloud. Complexities such as job scheduling should be hidden from user.
+
+## Training Job
+
+A training job will be created once user asks Paddle cloud to train a model. The training job is made up of different processes that collaboratively consume data and produce a trained model. There are three kinds of processes:
+
+- Master process
+- Trainer process
+- Parameter server process
+
+One training job will only have one master process, typically multiple trainer processes and parameter server processes. Their relation is illustrated in the following graph:
+
+<img src="src/paddle-model-sharding.png" height="400"/>
+
+### Master Process
+
+The master process will:
+
+- Do [parameter server selection](#parameter-server-selection).
```

Comment:
Do we need parameter server selection? I'd thought that we could have an easy solution -- once a parameter server starts, it registers itself to the etcd directory `/paddle/<job_name>/pservers/<i>` where `<i>` is the minimum available ID.  Once there are enough number of parameter server processes (and trainer processes) registered, the master process can start dispatching tasks.